openlibrary/catalog/get_ia.py

```diff
@@ -26,16 +26,17 @@ class NoMARCXML(IOError):
     pass
 
 
-def urlopen_keep_trying(url):
+# TODO: there are few other modules that call this, refactor to rename later to request_keep_trying
+def urlopen_keep_trying(url, headers=None):
+    """Tries to request the url three times, raises HTTPError if 403, 404, or 416.  Returns a requests.Response"""
     for i in range(3):
         try:
-            f = urllib.request.urlopen(url)
-            return f
-        except urllib.error.HTTPError as error:
-            if error.code in (403, 404, 416):
+            resp = requests.get(url, headers=headers)
+            resp.raise_for_status()
+            return resp
```

Comment:
https://requests.readthedocs.io/en/master/api/#requests.Response

Please make sure that all callers are prepared for that change.

Reply:
I did change all the callers, yes.